src/compat/db_vdb.c

```diff
@@ -677,7 +677,12 @@ db_compile_statement_local (DB_SESSION * session)
    */
   if (PT_IS_DBLINK_DML_QUERY (statement))
     {
-      ;
+      /*
+         The remote DML query can not use the cached plan,
+         because the atuo-variable holder can not be composed during compiling.
+         Instead the remote query is generated by parser_print_tree.
+       */
+      statement->flag.recompile = 1;
```

Comment:
I think there should be an example in the comments of the code or in the comments of issue CBRD-24501 for understanding.

Reply:
Ok, I will comment on the JIRA issue CBRD-24501.

Reply:
I saw it. Thank you.